src/Orleans/Streams/Core/IOnSubscriptionActioner.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Orleans.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Streams.Core
+{
+    public interface IOnSubscriptionActioner
+    {
+        Task OnAdd<T>(StreamSubscriptionHandle<T> handle);
```

Comment:
OnAdd sounds like it's being told to add, when it's being notified that a subscription has already been added.
OnNewSubscription?  OnSubscriptionAdded?  OnSubscribed?